### PR TITLE
Added shk9 macropad keyboard

### PR DESCRIPTION
**Pull request by samsamm777**
Added support for the SHK9 3x3 macropad. Has VIA Support

## Description

Added support for the SHK9 3x3 macropad.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Comment by samsamm777** [comment]
@drashna How often do PRs get merged?

**Comment by tzarc** [comment]
> 
> 
> @drashna How often do PRs get merged?

When they're ready, basically. Needs two collaborators to approve.
I'd ask you to go through each of the line items in the PR checklist document -- you've ticked it as completed but things like having a `DESCRIPTION` listed in your code is still present. No merge will occur until they get addressed.

File: keyboards/shk9/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    [_BASE] = LAYOUT(
+        KC_1, KC_2, KC_3,
+        KC_4, KC_5, KC_6,
+        KC_7, KC_8, KC_9
+    )
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    return true;
+}
```
Comment (tzarc) [inline keyboards/shk9/keymaps/default/keymap.c:34]:
```suggestion
```

File: keyboards/shk9/config.h [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x5348
+#define PRODUCT_ID   0x4B39
+#define DEVICE_VER   0x0001
+#define MANUFACTURER Super Human
+#define PRODUCT      SHK9
+#define DESCRIPTION  3x3 Keyboard
```
Comment (tzarc) [inline keyboards/shk9/config.h:28]:
```suggestion
```

File: keyboards/shk9/rules.mk [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+COMMAND_ENABLE = yes         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+LINK_TIME_OPTIMIZATION_ENABLE = yes
```
Comment (tzarc) [inline keyboards/shk9/rules.mk:23]:
```suggestion
LTO_ENABLE = yes
```

File: keyboards/shk9/rules.mk [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+FAUXCLICKY_ENABLE = no
```
Comment (tzarc) [inline keyboards/shk9/rules.mk:24]:
```suggestion
```

File: keyboards/shk9/config.h [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+//#    define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//#    define RGBLIGHT_EFFECT_RGB_TEST
+//#    define RGBLIGHT_EFFECT_ALTERNATING
+/*== customize breathing effect ==*/
+/*==== (DEFAULT) use fixed table instead of exp() and sin() ====*/
+//#    define RGBLIGHT_BREATHE_TABLE_SIZE 256      // 256(default) or 128 or 64
+/*==== use exp() and sin() ====*/
+//#    define RGBLIGHT_EFFECT_BREATHE_CENTER 1.85  // 1 to 2.7
+//#    define RGBLIGHT_EFFECT_BREATHE_MAX    255   // 0 to 255
+//#endif
```
Comment (tzarc) [inline keyboards/shk9/config.h:92]:
```suggestion
```

File: keyboards/shk9/config.h [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+/*
+ * Keyboard Matrix Assignments
+ *
+ * Change this to how you wired your keyboard
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
```
Comment (tzarc) [inline keyboards/shk9/config.h:42]:
```suggestion
```

File: keyboards/shk9/config.h [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+/* disable debug print */
+//#define NO_DEBUG
+
+/* disable print */
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
```
Comment (tzarc) [inline keyboards/shk9/config.h:145]:
```suggestion
```

File: keyboards/shk9/config.h [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+
+/* Bootmagic Lite key configuration */
+//#define BOOTMAGIC_LITE_ROW 0
+//#define BOOTMAGIC_LITE_COLUMN 0
```
Comment (tzarc) [inline keyboards/shk9/config.h:153]:
```suggestion
```

**Review by tzarc — CHANGES_REQUESTED** [review]

**Comment by samsamm777** [comment]
> > @drashna How often do PRs get merged?
> 
> When they're ready, basically. Needs two collaborators to approve.
> I'd ask you to go through each of the line items in the PR checklist document -- you've ticked it as completed but things like having a `DESCRIPTION` listed in your code is still present. No merge will occur until they get addressed.

Thanks @tzarc. I've addressed the changes you requested.

File: keyboards/shk9/info.json [hunk 9]
```diff
@@ -0,0 +1,23 @@
+{
+    "keyboard_name": "SHK9",
+    "url": "http://superhuman.website",
+    "maintainer": "Sam Hudson",
+    "width": 3,
+    "height": 3,
+    "lighting": "none",
+    "layouts": {
+        "LAYOUT_3x3": {
```
Comment (drashna) [inline keyboards/shk9/info.json:9]:
```suggestion
        "LAYOUT_ortho_3x3": {
```

File: keyboards/shk9/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_3x3( /* Base */
```
Comment (drashna) [inline keyboards/shk9/keymaps/default/keymap.c:19]:
```suggestion
    [0] = LAYOUT_ortho_3x3( /* Base */
```

File: keyboards/shk9/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+        _______, _______, _______,
+        _______, _______, _______,
+        _______, _______, _______
+    ),
+    [2] = LAYOUT_3x3(
+        _______, _______, _______,
+        _______, _______, _______,
+        _______, _______, _______
+    ),
+    [3] = LAYOUT_3x3(
```
Comment (drashna) [inline keyboards/shk9/keymaps/via/keymap.c:34]:
```suggestion
    [0] = LAYOUT_ortho_3x3( /* Base */
        KC_1, KC_2, KC_3,
        KC_4, KC_5, KC_6,
        KC_7, KC_8, KC_9
    ),
    [1] = LAYOUT_ortho_3x3(
        _______, _______, _______,
        _______, _______, _______,
        _______, _______, _______
    ),
    [2] = LAYOUT_ortho_3x3(
        _______, _______, _______,
        _______, _______, _______,
        _______, _______, _______
    ),
    [3] = LAYOUT_ortho_3x3(
```

File: keyboards/shk9/shk9.h [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+/* This is a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT_3x3( \
```
Comment (drashna) [inline keyboards/shk9/shk9.h:29]:
```suggestion
#define LAYOUT_ortho_3x3( \
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/shk9/readme.md [hunk 13]
```diff
@@ -0,0 +1,23 @@
+# shk9
+
+![shk9](https://i.imgur.com/FluRgiH.jpg)
```
Comment (noroadsleft) [inline keyboards/shk9/readme.md:3]:
```suggestion
![shk9](https://i.imgur.com/FluRgiHl.jpg)
```

File: keyboards/shk9/readme.md [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+
+The SHK9 is a tiny 3x3 macropad! Fully configurable with via.
+
+* Keyboard Maintainer: [Sam Hudson](https://github.com/samsamm777)
+* Hardware Supported: *The PCBs, controllers supported*
```
Comment (noroadsleft) [inline keyboards/shk9/readme.md:8]:
```suggestion
* Hardware Supported: SHK9 PCB (ATmega32u4)
```

File: keyboards/shk9/readme.md [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+
+Flashing example for this keyboard:
+
+    make shk9:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader Mode
+
+To reset the board into bootloader mode, remove the pcb from the case buy removing keycaps, and unscrewing the 3 screws. Then (while plugged in) press the switch on the underside of the PCB for a few seconds. The board is now in bootloader mode.
```
Comment (noroadsleft) [inline keyboards/shk9/readme.md:23]:
```suggestion
## Bootloader Mode

To reset the board into bootloader mode, remove the pcb from the case by removing keycaps, and unscrewing the 3 screws. Then (while plugged in) press the switch on the underside of the PCB for a few seconds. The board is now in bootloader mode.

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/shk9/rules.mk [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes         # Commands for debug and configuration
```
Comment (noroadsleft) [inline keyboards/shk9/rules.mk:14]:
```suggestion
COMMAND_ENABLE = yes        # Commands for debug and configuration
```

File: keyboards/shk9/rules.mk [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
```
Comment (noroadsleft) [inline keyboards/shk9/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes           # USB Nkey Rollover
```

File: keyboards/shk9/rules.mk [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+UNICODE_ENABLE = no
```
Comment (noroadsleft) [inline keyboards/shk9/rules.mk:23]:
```suggestion
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/shk9/keymaps/default/keymap.c [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_ortho_3x3( /* Base */
+        KC_1, KC_2, KC_3,
+        KC_4, KC_5, KC_6,
+        KC_7, KC_8, KC_9
```
Comment (drashna) [inline keyboards/shk9/keymaps/default/keymap.c:22]: This is mostly nitpicky, to maintain the 7 character spacing
```suggestion
        KC_1,    KC_2,    KC_3,
        KC_4,    KC_5,    KC_6,
        KC_7,    KC_8,    KC_9
```

File: keyboards/shk9/keymaps/via/keymap.c [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_ortho_3x3( /* Base */
+        KC_1, KC_2, KC_3,
+        KC_4, KC_5, KC_6,
+        KC_7, KC_8, KC_9
```
Comment (drashna) [inline keyboards/shk9/keymaps/via/keymap.c:22]:
```suggestion
        KC_1,    KC_2,    KC_3,
        KC_4,    KC_5,    KC_6,
        KC_7,    KC_8,    KC_9
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/shk9/keymaps/default/keymap.c (hunk 19 above)
Comment (samsamm777) [reply keyboards/shk9/keymaps/default/keymap.c:22]: I would only agree with this if there were entries with 7 characters, but given there are not i think this spacing is not neccessary.

File: keyboards/shk9/keymaps/via/keymap.c (hunk 20 above)
Comment (samsamm777) [reply keyboards/shk9/keymaps/via/keymap.c:22]: This just looks untidy and does nothing for readability.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**